test/layout.ts

```diff
@@ -1048,4 +1072,59 @@ describe("Layout Tests", () => {
       done();
     }
   });
+
+  it("checks that input value is masked on insert if the config is set to show text and mask attribute is applied", (done: DoneFn) => {
+    let observer = new MutationObserver(callback);
+    observer.observe(document, { childList: true, subtree: true });
+
+    config.showText = true;
+
+    // Add a node to the document and observe Clarity events
+    let stopObserving = observeEvents(eventName);
+    let input = document.createElement("input");
+    let valueString = "value";
+    let maskedValueString = mask(valueString);
+    input.setAttribute("value", valueString);
+    input.setAttribute(ForceMaskAttribute, "true");
+    document.body.appendChild(input);
+
+    function callback() {
+      observer.disconnect();
+      let events = stopObserving();
+      assert.equal(events.length, 1);
+      assert.equal(events[0].state.tag, "INPUT");
+      assert.equal(events[0].state.action, Action.Insert);
+      assert.equal(events[0].state.attributes["value"], maskedValueString);
+      done();
+    }
+  });
+
+  it("checks that text value is masked when the config is set to show text and parent mask attribute is applied", (done: DoneFn) => {
```

Comment:
>that [](start = 13, length = 4)

Right now if a mutation adds "data-clarity-mask" child elements wouldn't automatically inherit the masking property. FYI. We can file it as an issue and come back.

Reply:
Added a test that currently fails and marked it as a bug

---
In reply to: [182854375](https://github.com/Microsoft/clarity-js/pull/80#discussion_r182854375) [](ancestors = 182854375)